### PR TITLE
[Intel MKL] Adding support for Depthwise 2D Convolution (fwd)

### PR DESCRIPTION
**Pull request by claynerobison**

### CODE REVIEW

File: tensorflow/core/kernels/mkl_conv_ops.h
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.h]: `filter` -> `filter shape (filter_dims)`
(Using just `filter` confused me with the actual filter tensor which is in TF layout.)

File: tensorflow/core/kernels/mkl_conv_ops.h
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.h]: Please correct the typo and add spaces for readability.
`in_depth*channel_muliplier` -> `in_depth * channel_multiplier`

This file also haven't mentioned `channel_multiplier` before, so please also explain that this means the `out_depth` of the filter shape.

File: tensorflow/core/kernels/mkl_conv_ops.h
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.h]: For future PRs, what do you think about adding an enum (for TF format indices) or getter functions? For example, `filter_shape.dim_size(TFFilterDims::in_depth)` or `GetInDepthFromTFFormat(filter_shape)`. It will be more readable and flexible for future changes.

File: tensorflow/core/graph/mkl_layout_pass_test.cc
Comment (penpornk) [inline tensorflow/core/graph/mkl_layout_pass_test.cc]: `DepthwiseConv2dNative` doesn't have the attribute `use_cudnn_on_gpu`. (See [its attribute page](https://www.tensorflow.org/api_docs/cc/struct/tensorflow/ops/depthwise-conv2d-native/attrs), versus Conv2D's [attribute page](https://www.tensorflow.org/api_docs/cc/struct/tensorflow/ops/conv2-d/attrs).) Maybe remove this line?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! Apart from the comments, the files have some formatting errors. Please also run `clang-format`.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by claynerobison** [comment]
@penpornk We've addressed all these issues internally, but we're waiting for https://github.com/tensorflow/tensorflow/pull/21483 to merge first.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by claynerobison** [comment]
@penpornk We're ready to merge this. Can you review/approve?

File: tensorflow/core/kernels/mkl_conv_ops.h
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.h]: The comment line above this line is short. Could you please put this line with the previous line? (And wrap the line again if it's too long). Also, a new typo at the end of the line: `channel_miltipler` should be `channel_multiplier`.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you very much for the changes and sorry for the delay! I found two minor typos in the comments. We should be ready to go after these corrections.

File: tensorflow/core/kernels/mkl_conv_ops.h
Comment (claynerobison) [reply tensorflow/core/kernels/mkl_conv_ops.h]: Not a problem, @penpornk. Thanks for reviewing.

**Review by penpornk — DISMISSED** [review]
Thank you very much again for the PR!

**Comment by rmlarsen** [comment]
@claynerobison I apologize for the long delay. Could I ask you to please rebase to resolve the conflict?

**Comment by claynerobison** [comment]
@rmlarsen We're preparing to submit a few more PRs for 1.13 that are going to cause more conflicts, so let's wait on this one.

**Comment by rmlarsen** [comment]
@claynerobison sounds good. Thanks.

**Comment by agramesh1** [comment]
@rmlarsen  the conflicts have been resolved. Thanks

**Review by penpornk — APPROVED** [review]
Sorry for my delay. I didn't notice that this needs re-approval. Thank you again for the PR!

**Comment by agramesh1** [comment]
Hi @penpornk , It looks like the testing is stuck waiting for internal safe approval. Is there anything we need to change to pass the review?

**Comment by penpornk** [comment]
@agramesh1 Rasmus is on vacation starting from today. @harshini-gadige is helping me pulling the PR in right now. Hopefully it can be merged soon.

**Comment by penpornk** [comment]
@agramesh1 Unfortunately, copybara was acting strange today and this PR wasn't imported correctly. I had to apply manual fix internally and so it needed one more reviewer (since I can't review my own changes). Waiting for review right now. I'll keep you updated.

**Pull request merged**
